From ed81a44e6faffd450d9ab779a7c5862a9bd7c773 Mon Sep 17 00:00:00 2001
From: Adam Borowski <kilobyte@angband.pl>
Date: Thu, 27 Jan 2022 12:08:56 +0100
Subject: [PATCH] test: don't print the address of a FILE after fclose

In pre-release gcc-12 that Fedora releases with, it triggers
-Werror=use-after-free despite us not dereferencing the pointer.
Its address is no longer useful for debugging anyway.
---
 src/test/unittest/ut_file.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/test/unittest/ut_file.c b/src/test/unittest/ut_file.c
index 26f5cbda6..266b86387 100644
--- a/src/test/unittest/ut_file.c
+++ b/src/test/unittest/ut_file.c
@@ -88,10 +88,8 @@ ut_fclose(const char *file, int line, const char *func, FILE *stream)
 {
 	int retval = os_fclose(stream);
 
-	if (retval != 0) {
-		ut_fatal(file, line, func, "!fclose: 0x%llx",
-			(unsigned long long)stream);
-	}
+	if (retval != 0)
+		ut_fatal(file, line, func, "!fclose");
 
 	return retval;
 }
-- 
2.34.1

